### DISCUSSION TITLE
AOSP 11.0 v312

### DISCUSSION THREAD

**Discussion by phhusson** [phhusson/treble_experimentations#1964] [General]
- VoLTE on Mediatek R vendors
- Under-display fingerprint comeback. Partial support on OnePlus Nord2.
- Double-tap to wake on OnePlus Nord2
- Fix fingerprint support on some OPPO/OP devices
- Fix securize while running Magisk in some cases (Thanks @hwti )

--- All those changes (except last one), are because I bought that OnePlus Nord2 from my own money. so if you like this changelog, don't forget to sponsor me

<hr /><em>This discussion was created from the release <a href='https://github.com/phhusson/treble_experimentations/releases/tag/v312'>AOSP 11.0 v312</a>.</em>

**Comment by innofeld** [comment]
Dear Pierre-Hugues HUSSON
Many thanks for your unconditional efforts
We are so thankful
I have BV9500-plus and BV9900 smartphones from BlackView company
Can you please guide us to install your Android 11 ROM on that phones?
Regards
Yours Sincerely

**Reply by realKarthikNair** [reply]
@innofeld  check out [this](https://github.com/phhusson/treble_experimentations/wiki/Blackview-BV9900) link for BV9900 and refer [this](https://github.com/phhusson/treble_experimentations/wiki/Blackview-BV9500-Plus) one out for BV9500 plus

**Comment by illinoisbutch** [comment]
Absolutely amazing work.. Snapdragon S20plus

**Comment by Mariyan** [comment]
How do you flash this on a Samsung phone? GSI flasher or fastbootd commands?

**Reply by realKarthikNair** [reply]
Fastboot isn't something you generally find on samsung devices,  (except on around a total of 4-5 devices from the S, Note and A  series)
Another way would be flashing through a custom recovery (if available) or on odin or Heimdall itself (if no custom recovery; note that flashing using odin is complicated if you're not really smart enough with Android and Samsung in general )
Also, if your Samsung device is having a super partition, things might be even more complicated.
Note : if it's a Samsung Qualcomm device, don't be pissed off if that doesn't boot; be confident if it's an exnyos plus you've managed to keep up the points above. About Mediatek Samsungs, i don't know much since i never got my hands on one of those.

Also refer to the wiki to know more about your specific device model ( if lucky enough, you'ff find link to an entire page explaining everything about how-to-install-gsi on your particular samsung device)

That's all i can help for now since you HAVE'NT mentioned your device model at the first place (such a shame)

**Reply by innofeld** [reply]
Hi,
my idea is to merge your existing image based on Android 11 (AOSP 11.0 v312), but with rely on Android 11 with Kernel 5.4.x LTS , and additional patches of Trusty TEE 5.4
You are experienced and talented in this sector, so I can give you an idea.
Please see my corrections:

A1. https://android.googlesource.com/kernel/common/+/refs/heads/android11-5.4-lts = this webpage contains Android 11 with Kernel 5.4.x LTS; so LTS means longterm support and best stability in kernel which is good advantage.

A2. https://android.googlesource.com/kernel/common/+/refs/heads/android-trusty-5.4 = this webpage contains Android Trusty with kernel 5.4.x

My idea is to create new image something like "AOSP 11.0 v312", but based on A1 with patches of A2
I think results will be very stable image, integrated with latest kernel version, and with more security.
I'm sure if you can create this image, it will be innovative, stable, agile and high integrity image to be installed on any smartphones.

Personally I have BV9500-plus smartphone from BlackView company
And I'm highly interested to try this image on my smartphone

Regards and Respect

**Reply by realKarthikNair** [reply]
Sorry I'm noob. I don't understand stuff you're talking.

**Reply by illinoisbutch** [reply]
> How do you flash this on a Samsung phone? GSI flasher or fastbootd commands?

I have Snapdragon S20 plus. It is bootloader unlocked used on At&t in United States. I used Chondoe Gsi flasher to deal with dynamic portions. This is the only gsi or treble that I have found that works well enough to use as daily driver. The battery life is amazing. Only thing missing for me is wifi calling since where I live data connection is horrible. But other than that..pure gold

**Reply by illinoisbutch** [reply]
(Dynamic partitions)

**Reply by Yilliee** [reply]
> Fastboot isn't something you generally find on samsung devices, (except on around a total of 4-5 devices from the S, Note and A series)
> Another way would be flashing through a custom recovery (if available) or on odin or Heimdall itself (if no custom recovery; note that flashing using odin is complicated if you're not really smart enough with Android and Samsung in general )
> Also, if your Samsung device is having a super partition, things might be even more complicated.
> Note : if it's a Samsung Qualcomm device, don't be pissed off if that doesn't boot; be confident if it's an exnyos plus you've managed to keep up the points above. About Mediatek Samsungs, i don't know much since i never got my hands on one of those.
> 
> Also refer to the wiki to know more about your specific device model ( if lucky enough, you'ff find link to an entire page explaining everything about how-to-install-gsi on your particular samsung device)
> 
> That's all i can help for now since you HAVE'NT mentioned your device model at the first place (such a shame)

Allow me to offer a correction, as he mentioned GSI Flasher and fastbootd, it should be easily understandable that he is talking about a dynamic device (Super Partition). All SAMSUNG dynamic devices have working fastbootd.

**Reply by Yilliee** [reply]
> How do you flash this on a Samsung phone? GSI flasher or fastbootd commands?

Fastbootd is indeed the recommended option.
But in case you have a twrp based custom recovery from which you don't wanna switch and fastbootd doesn't work in it you can use GSI Flasher too. There is one recovery patcher that I made named DynaPatch that just patches your recovery and allows flashing images directly using the twrp flash image option but as of right now it only works on limited devices and is still WiP.

**Comment by innofeld** [comment]
Dear Pierre-Hugues HUSSON
Please create an image by this OS, Trusty TEE
https://source.android.com/security/trusty#application_services
https://android.googlesource.com/kernel/common/+/refs/heads/android12-trusty-5.10
And LTS version of Android 11: https://android.googlesource.com/kernel/common/+/refs/heads/android11-5.4-lts
Regards
Innofeld

**Reply by phhusson** [reply]
trusty isn't generic, how would I do that...?
for the second one, are you asking me for a gki?

**Reply by innofeld** [reply]
Hi,
my idea is to merge your existing image based on Android 11 (AOSP 11.0 v312), but with rely on Android 11 with Kernel 5.4.x LTS , and additional patches of Trusty TEE 5.4
You are experienced and talented in this sector, so I can give you an idea.
Please see my corrections:

  A1.    https://android.googlesource.com/kernel/common/+/refs/heads/android11-5.4-lts    =    this webpage contains Android 11 with Kernel 5.4.x LTS; so LTS means longterm support and best stability in kernel which is good advantage.

  A2.    https://android.googlesource.com/kernel/common/+/refs/heads/android-trusty-5.4     =    this webpage contains Android Trusty with kernel 5.4.x

My idea is to create new image something like "AOSP 11.0 v312", but based on A1 with patches of A2
I think results will be very stable image, integrated with latest kernel version, and with more security.
I'm sure if you can create this image, it will be innovative, stable, agile and high integrity image to be installed on any smartphones.

Personally I have BV9500-plus smartphone from BlackView company
And I'm highly interested to try this image on my smartphone

Regards and Respect

**Reply by phhusson** [reply]
Why would your device boot A1 kernel at all?

**Reply by innofeld** [reply]
Thanks for your reply
Because it has LTS and also it's kernel 5.4.x which is newer that 4.4x
LTS means more support and more bug-fixes which means more stable and more secure devices
Regards

**Reply by innofeld** [reply]
Dear Pierre
Will you provide an unique image like my idea ?
Let's do it
Regards

**Reply by innofeld** [reply]
Dear Pierre
In fact, I need to install Android 11 + Kernel 5.4.x on my smartphone, "BV9500-plus"
In this regard, I need your existing "AOSP 11.0 v312"; but with this following kernel, of course if possible
    https://android.googlesource.com/kernel/common/+/refs/heads/android11-5.4-lts
Kernel 5.4 LTS is extremely stable and reliable, more secure and has better performance in total.
With Sincere Respect
INNOFELD

**Comment by gimmebuckets** [comment]
Hello,  please excuse my ignorance.  I have a Samsung S20 5G (986U1) bootloader unlocked, firmware one ui 3.0.  I am trying to find a snapdragon rom with out google.  I am unfamiliar with Github and the file systems.  How do I determine which file to download for my phone?  Will this rom run with microG or something to spoof the play store?

Thanks

Kenny

**Reply by Yilliee** [reply]
You can use treble info app from play store to check which version to use.
As for MicroG, the foss varaint had it pre-included if I remeber correctly, tho I might be wrong.

**Comment by Paritosh-Kumar-Tripathi** [comment]
Dear Pierre-Hugues HUSSON
Many thanks for your unconditional efforts
We are so thankful
I have a Panasonic Eluga I7 EE smartphone from Panasonic company
Can you please guide us to install your Android 11 Go Edition ROM on this phone?
Regards
Yours Sincerely

**Comment by innofeld** [comment]
Dear Pierre
In fact, I need to install Android 11 + Kernel 5.4.x on my smartphone, "BlackView BV9500-plus"
In this regard, I need your existing "AOSP 11.0 v312"; but with this following kernel, of course if possible
    https://android.googlesource.com/kernel/common/+/refs/heads/android11-5.4-lts
Kernel 5.4 LTS is extremely stable and reliable, more secure and has better performance in total.
With Sincere Respect
INNOFELD

**Comment by SmileYzn** [comment]
Thanks for work! :pray: 

Working nice on Motorola One Fusion with AOSP 11.0 v312 (Open Gapps version A/B)

Ps. I think it is XT-2073 128gb.
Anyway magisk path boot.img not working for now :|

**Comment by MantoOfficial** [comment]
> * VoLTE on Mediatek R vendors
> * Under-display fingerprint comeback. Partial support on OnePlus Nord2.
> * Double-tap to wake on OnePlus Nord2
> * Fix fingerprint support on some OPPO/OP devices
> * Fix securize while running Magisk in some cases (Thanks @hwti )
> 
> --- All those changes (except last one), are because I bought that OnePlus Nord2 from my own money. so if you like this changelog, don't forget to sponsor me
> 
> _This discussion was created from the release [AOSP 11.0 v312](https://github.com/phhusson/treble_experimentations/releases/tag/v312)._

AOSP 11.0.v312# reboot three or four time and boot successful
Sir please need to fix boot android 8.1 Oreo (Redmi Go tiare Devices) arm a only

**Comment by MantoOfficial** [comment]
Sir please fix boot issues in 
Device 👉Redmi Go Tiare(arm a only 8.1 Oreo)
Issues In AOSP 11.0.v312# device reboot and reboot  5-6 time and and boot success very late

**Reply by JonnyTech** [reply]
> Sir please fix boot issues in
> Device 👉Redmi Go Tiare(arm a only 8.1 Oreo)
> Issues In AOSP 11.0.v312# device reboot and reboot 5-6 time and and boot success very late

@MantoOfficial I am running AOSP11 v312 on my Redmi Go without issue, can you explain your issue fully please?

**Comment by MantoOfficial** [comment]
Some time when I'm turn on device my devices reboot aoutmatic 4-5 Time and
boot success some time
And how to remove search bar in home screen I remove searchbox but search
bar not gone

**Reply by JonnyTech** [reply]
> Some time when I'm turn on device my devices reboot aoutmatic 4-5 Time and boot success some time And how to remove search bar in home screen I remove searchbox but search bar not gone

Flash full stock and check if that works. I suspect that you flashed magisk hence the reboot issue. Flashing stock including boot.img then gsi should fix reboots.

To remove search bar you need to install another launcher, you cannot remove the searchbar from aosp launcher.
